### ISSUE TITLE
Add Earlystopping function to train.py of the speech commands API

### ISSUE THREAD

**Issue by alex83803** [tensorflow/tensorflow#37302]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): Tensorflow 2.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**

As you know, the amount of training epochs is hard coded in the train.py file of the API. the 18000 epochs are perfect if we train the model with the official dataset. 

**Will this change the current api? How?**

Adding a Earlystop function will prevent under or overfitting regardless which dataset will be trained. 

**Who will benefit with this feature?**

I would really apreciate if one of your could emplement a version of train.py with a Early stop flag. 
This would be a plus for the API and of big help for my use case.

**Any Other info.**

If you guys could give me a hint about how to add this feature to the existing API, I'd be happy to share my work.

Thank you very much 

Kind regards

**Comment by MeghnaNatraj** [comment]
We're porting this example to TF2.0 completely and will ensure this is factored in. I have personally wanted this feature as well, and it's one of our requirements for porting it. Is this a blocker for you right now?

**Comment by alex83803** [comment]
Hi Megna,

these are really great news. This feature is a great plus if we want to add a smooth workflow that consists on adding new labels and train.

Currently I have to adjust the number of epochs manuals depending on my training results and the EarlyStopping function would save me a lot of time. This is kind of a Blocker because my workflow is not completely automated :-).

I a really looking forward to the 2.0 version of this API. When do you think that EarlyStopping will be implemented?

Best thanks

**Comment by MeghnaNatraj** [comment]
@alex83803 we will not be updating the TF1 code, however you can follow the updates in #41310 for the TF2 version. Closing this issue.

**Comment by alex83803** [comment]
Hey Meghna,

No Problem this is what we had discussed.

When do you think the tensorflow 2.0 version of the speech recognition framework will be ready?

Kind regards

Alex

**Comment by MeghnaNatraj** [comment]
This is an additional feature, so it's not high priority - could take weeks/few months! Subscribe to #41310 for updates.

**Comment by alex83803** [comment]
I understand ;-).

I’ll keep tuned. 

Thanks a lot 



Inh. Alexandre Davin
Leopoldstraße 50
80802 München
08938338400
